go.mod

```diff
@@ -29,7 +29,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	github.com/tendermint/tendermint v0.34.29
 	github.com/tendermint/tm-db v0.6.7
-	golang.org/x/exp v0.0.0-20231206192017-f3f8817b8deb
```

Comment:
@smuu Out of curiosity, why this change is needed? I backported it from the other PR https://github.com/celestiaorg/celestia-app/pull/3493/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R32